pkg/apis/migration/migration.go

```diff
@@ -374,18 +392,24 @@ func (volumeMigration *volumeMigration) registerVolume(ctx context.Context, volu
 				BackingDiskUrlPath: backingDiskURLPath,
 			},
 		}
-		log.Debugf("vSphere CNS driver registering volume %q with create spec %+v", volumePath, spew.Sdump(createSpec))
+		if storagePolicyID != "" {
+			profileSpec := &vim25types.VirtualMachineDefinedProfileSpec{
+				ProfileId: storagePolicyID,
+			}
+			createSpec.Profile = append(createSpec.Profile, profileSpec)
+		}
+		log.Debugf("vSphere CNS driver registering volume %q with create spec %+v", volumeSpec.VolumePath, spew.Sdump(createSpec))
 		volumeID, err = (*volumeMigration.volumeManager).CreateVolume(ctx, createSpec)
 		if err != nil {
-			log.Warnf("failed to register volume %q with error %+v", volumePath, err)
+			log.Warnf("failed to register volume %q with createSpec: %v. error: %+v", volumeSpec.VolumePath, createSpec, err)
 		} else {
 			break
 		}
 	}
 	if volumeID != nil {
-		log.Infof("successfully registered volume %q as container volume with ID: %q", volumePath, volumeID.Id)
+		log.Infof("successfully registered volume %q as container volume with ID: %q", volumeSpec.VolumePath, volumeID.Id)
 	} else {
-		msg := fmt.Sprintf("registration failed for volumePath: %q", volumePath)
+		msg := fmt.Sprintf("registration failed for volumePath: %q", volumeSpec.VolumePath)
```

Comment:
Can we add more context like Storage Policy name/Id?

Reply:
In the message above we print all details with

```
log.Warnf("failed to register volume %q with createSpec: %v. error: %+v", volumeSpec.VolumePath, createSpec, err)
```

with the contextual tracing, we can get all details.

changing this to

```
msg := fmt.Sprintf("registration failed for volumeSpec: %v", volumeSpec)
```